aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -2243,6 +2243,74 @@ Tensor nonzero_cpu(const Tensor& self) {
   return result;
 }
 
+Tensor& nonzero_static_out_cpu(
+    const Tensor& self,
+    int64_t size,
+    int64_t fill_value,
+    Tensor& result) {
+  // Sanity check on input
+  TORCH_CHECK(size >= 0, "'size' must be an non-negative integer");
+  int64_t ndim = self.dim();
+  TORCH_CHECK(ndim > 0, "Rank must be positive");
+  TORCH_CHECK(
+      result.scalar_type() == kLong,
+      "nonzero_static: Expected out tensor to have scalar type Long "
+      "but got scalar type",
+      result.scalar_type());
+  TORCH_CHECK(
+      result.dim() == 2,
+      "nonzero_static: Expected out tensor to be a 2D tensor but got a ",
+      result.dim(),
+      "D tensor");
```

Comment:
This is not entirely consistent with the prevailing convention for out= variants. In particular, when you pass a zero size out, PyTorch traditionally resizes it to the requested output size. Use `at::native::resize_output` to get the correct behavior in these cases.

Reply:
@ezyang  Just to check, do we want resize only when passing a zero size out, or whenever the size is doesn't exactly meet the expected output size, e.g. larger or smaller?